### ISSUE TITLE
Consider adding a ParamConverter to support array parameters in REST endpoints

### ISSUE THREAD

**Issue by tjquinno** [helidon-io/helidon#2220]
## Environment Details
* Helidon Version: 2.x
* Helidon SE or Helidon MP MP
* JDK version: n/a
* OS: n/a
* Docker version (if applicable): n/a

----------

## Problem Description
If a JAX-RS endpoint declares a parameter as an array (see example below), then Jersey throws an exception (see below).

This non-standard construct is/was used in the MP Metrics 2.3 optional TCK (and as a result temporarily blocked #2140).

This seems _not_ to be a bug in Helidon, because the JAX-RS spec does not mention, let alone mandate, support for array parameters. It is almost certainly a bug in the MP Metrics 2.3 TCK that uses this construct which might be supported by some JAX-RS and/or MP implementations.

We could consider adding a `ParamConverter` and a provider for it to Helidon MP to support array parameters to REST endpoints.

## Steps to reproduce
Create a Helidon MP app with an endpoint that declares an array parameter (see below).

Jersey throws an exception similar to this because it does not support arrays intrinsically: (newlines and indentation inserted manually for readability)
```
org.glassfish.jersey.server.model.ModelValidationException: 
Validation of the application resource model has failed during application initialization.
[[FATAL] No injection source found for a parameter of type 
public java.lang.String io.helidon.microprofile.metrics.HelloWorldResource.getArrayParam(java.lang.String[])
at index 0.; 
source='ResourceMethod{httpMethod=GET, consumedTypes=[], producedTypes=[text/plain],
  suspended=false, suspendTimeout=0, suspendTimeoutUnit=MILLISECONDS, invocable=Invocable{
    handler=ClassBasedMethodHandler{
      handlerClass=class io.helidon.microprofile.metrics.HelloWorldResource, 
      handlerConstructors=[org.glassfish.jersey.server.model.HandlerConstructor@2cd388f5]},
    definitionMethod=public java.lang.String
      io.helidon.microprofile.metrics.HelloWorldResource.getArrayParam(java.lang.String[]), 
      parameters=[Parameter [type=class [Ljava.lang.String;, source=names, defaultValue=null]],
       responseType=class java.lang.String},
   nameBindings=[]}']
```

Example endpoint:
```
    @GET
    @Path("withNames")
    @Produces(MediaType.TEXT_PLAIN)
    public String getArrayParam(@QueryParam("names") String[] names) {
        return Arrays.stream(names)
                .collect(Collectors.joining("and", "Hello World, ", "!"));
    }
```

Replacing the `String[]` with `List<String>` allows Jersey to use its built-in support for `List` parameters and therefore works.

**Comment by tjquinno** [comment]
One possibility: Helidon MP supplies its own `ParamConverter` for arrays (or at least for arrays of primitives and selected Java-provided classes: String, Double, etc.

**Comment by tjquinno** [comment]
Laird and I have separately established that the JAX-RS spec does not mandate support for arrays. 

I am checking to make sure that the MP metrics TCK does not include a `ParamConverter` that should handle this that somehow our test framework is ignoring.

**Comment by tjquinno** [comment]
There are four classes in the optional part of the MP metrics 2.3 TCK, and none of them include a `ParamConverter` or a provider for one. So it would seem that the TCK assumes a feature in the JAX-RS implementation that is not part of the spec.

**Comment by tjquinno** [comment]
I created a topic about this in the MP Google group yesterday. So far a few views but no replies. I will probably create an issue against the MP metrics TCK early next week if not sooner.

**Comment by tjquinno** [comment]
I have opened this MP Metrics bug against the optional part of the 2.3 TCK: https://github.com/eclipse/microprofile-metrics/issues/601

**Comment by spericas** [comment]
That's correct, no support for arrays in Jakarta REST. TCK test is not valid without a param converter.
